website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -65,6 +65,43 @@ The most common types are:
  - `Gauge` Value can be increased and decreased as needed.
 5. `_total` suffix should be used for accumulating count. If your metrics has labels with high cardinality, like `pod`/`container` it usually means that you can aggregate it more, thus it will not require `_total` suffix.
 
+#### Recommended Metrics
+
+{{% alert title="Warning" color="warning" %}}
+The following operator health metric is new and may be subjected to change 
+{{% /alert %}}
+* `<operator-name-prefix>_operator_health_status` - The metric reports the operator health status. It should be reported by the operator itself based on the internal operator validations.  
+  This metric allows the cluster users, to view all operators health statuses with a single metric and improves the overall cluster observability.
+
+  In order to maintain consistency, it is recommended to use the following health statuses as the metric values:
+  * `0` - Indicates that the operator is healthy and working as expected.
+  * `1` - Indicates that the operator has some issues that needs to be addressed and can potentially lead to loss of functionality.
+  * `2` - Indicates that the operator is unhealthy and there is a loss of functionality that should be addressed.
+
+  Metric Type: `Gauge`
+
+  Metric labels:
+  * `name` - The operator name [Recommended].
+  
+  Metric Help:
+  * Indicates whether the operator health status is healthy (0), warning (1) or critical (2).
+
+  **Example:**  
+```
+	desc := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name:      "memcached_operator_health_status",
+			Help:      "Indicates whether the operator health status is healthy (0), warning (1) or critical (2).",
+		},prometheus.Labels{"name": "Memcached"})
+```
+
+  **Query:** To view all operators health statuses with a single query, users can use the following PromQL query, `{__name__=~".*operator_health_status"}`.
```

Comment:
```suggestion
  **Query:** To view all operators health statuses with a single query, users can use the following PromQL query, `{__name__="operator_health_status"}`.
```

Reply:
This is incorrect. The query I wrote is the correct one and it will return all the metrics for all the operators that will report the metric with the `operator_health_status` suffix.

Reply:
BTW, My first proposal was 1 metric name for all operators. 
But I changed this proposal based on:
1. We recommend in the best practices to add the operator name prefix for all of the operator custom metrics.
This helps to track where the metrics was exposed from which us important when you want to get a better look at the code that generate it. I believe this recommendation is still valid also in this case.
2. Prometheus state that it doesn't handle having different Help text for that same metric. It is reasonable to assume that there would be slight differences in the Help text when the same metric name is used by different opreators.

Reply:
> This helps to track where the metrics was exposed from which is important when you want to get a better look at the code that generated it.

I agree with this, but once again using controller-runtime as an example, the prefix isn't the operator name, it's the library that actually implements the metric. See https://github.com/kubernetes-sigs/controller-runtime/blob/c304e7ec2ee7879f53ce69c12a7aa133853fccdb/pkg/internal/controller/metrics/metrics.go#L25-L64

I would think we'd implement these in operator-lib and use one of these as the prefix:
- `operator_lib_`
- `operator_sdk_`

This piece of the discussion is another reason I think we should go ahead and immediately write a library that implements this proposal, and simultaneously review and comment on that. If we did that, it would immediately bring to the fore all of these gray areas about what actually originates where, who owns what, what can be tweaked and what can't by the operator author, what the cluster admin can rely on, etc.